app/uk/gov/hmrc/nisp/config/ApplicationConfig.scala

```diff
@@ -75,4 +81,39 @@ object ApplicationConfig extends ApplicationConfig with ServicesConfig {
   override val ivUpliftUrl: String = configuration.getString(s"identity-verification-uplift.host").getOrElse("")
   override val ggSignInUrl: String = configuration.getString(s"government-gateway-sign-in.host").getOrElse("")
   override val twoFactorUrl: String = configuration.getString(s"two-factor.host").getOrElse("")
+  override lazy val breadcrumbServiceUrl: String = configuration.getString(s"breadcrumb-service.url").getOrElse("")
+
+  override lazy val perTaxFrontEndUrl: String  = configuration.getString(s"pertax-frontend.url").getOrElse("")
+  lazy val pertaxServiceUrl = s"$perTaxFrontEndUrl/"
+  val initialBreadCrumbList = List((URLEncoder.encode(Messages("nisp.breadcrumb.account"),"UTF-8"), pertaxServiceUrl))
+  lazy val mainContentHeaderPartialUrl = s"$breadcrumbServiceUrl/integration/main-content-header"
+
+  private[config] def buildBreadCrumb(request: Request[_]): List[(String, String)] = {
+    val links = Map(
+      "account" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.pension"), "UTF-8"), routes.AccountController.show().url)),
+      "nirecord" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.nirecord"), "UTF-8"), routes.NIRecordController.showGaps().url)),
+      "voluntarycontribs" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.nirecord"), "UTF-8"), routes.NIRecordController.showVoluntaryContributions().url)),
+      "gapsandhowtocheck" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.nirecord"), "UTF-8"), routes.NIRecordController.showGapsAndHowToCheckThem().url)),
+      "exclusion" -> ((URLEncoder.encode(Messages("nisp.breadcrumb.account"), "UTF-8"), pertaxServiceUrl))
+    )
+    try {
+      val items = request.path.split("/").filter(!_.isEmpty).map(links.get).toList
+      initialBreadCrumbList ::: items.flatten
+    } catch {
+      case e: NoSuchElementException => Nil
+    }
+  }
+
+  def generateHeaderUrl() (implicit request:Request[_], user: NispUser): String = {
+    val userName = if(Some(user.name).isDefined) {
+      user.name.get
```

Comment:
You need to handle properly when an option is not defined. It is simpler to use a getOrElse